go/vt/vterrors/code.go

```diff
@@ -120,6 +120,8 @@ var (
 	VT14004 = errorWithoutState("VT14004", vtrpcpb.Code_UNAVAILABLE, "cannot find keyspace for: %s", "The specified keyspace could not be found.")
 	VT14005 = errorWithoutState("VT14005", vtrpcpb.Code_UNAVAILABLE, "cannot lookup sidecar database for keyspace: %s", "Failed to read sidecar database identifier.")
 
+	VT15001 = errorWithNoCode("VT15001", "session invalidated: close/reopen connection if applicable: %s", "This error means that the opened transaction should be closed by the application and re-opened.")
```

Comment:
"session invalidated" may be misleading
- if you are in a transaction, that needs to be closed, and a new transaction opened.
- if you are not in a transaction, a retry is probably what you need to do.
@harshit-gangal wdyt?

Reply:
It is the transaction which no longer exists. They can continue to use the same connection. 
"rollback", followed by a retry is fine.
outside of transaction, retry is enough as Deepthi pointed.
All 3 error message should have different error code as they happen at different point in time of PRS/ERS